utils.py

```diff
@@ -0,0 +1,81 @@
+from requests import get
+from requests.exceptions import ConnectionError
+from functools import wraps
+from sanic.exceptions import *
+from nbconvert import HTMLExporter
+from nbformat.v4 import to_notebook
+
+
+# Main conversion function accepting ipynb json and returning an HTML representation
+# As this may be slow, it is defined as a coroutine so it can be awaited
+async def perform_notebook_conversion(notebook_json):
+    # Get the notebook json into a NotebookNode object that nbconvert can use
+    nb = to_notebook(notebook_json)
+
+    # set up a default nbconvert HTML exporter and run the conversion
+    html_exporter = HTMLExporter()
+    (nb_html, resources_dict) = html_exporter.from_notebook_node(nb)
+    return nb_html
+
+
+# Authorization decorator
+# Decorated routes will first check the request to see if the caller is authorized to perform the operation
+# taken from example in Sanic documentation
+# https://sanic.readthedocs.io/en/latest/sanic/decorators.html
+def authorized(sam_root):
+    def decorator(f):
+        @wraps(f)
+        async def decorated_function(request, *args, **kwargs):
+            # Check the Terra authorization service SAM for user auth status
+            user_is_authorized = await check_sam_authorization(request, sam_root)
+            if user_is_authorized:
+                # run the handler method and return the response
+                response = await f(request, *args, **kwargs)
+                return response
+            else:
+                raise Forbidden
+        return decorated_function
+    return decorator
+
+
+# Query Terra's authorization service SAM to determine user authorization status. Return auth status boolean
+async def check_sam_authorization(request, sam_root):
```

Comment:
Also, I believe you can indicate private methods in python by prefixing method names with `__`

Reply:
Nice! I never knew that. I'll update this to make use of it